DirectProgramming/DPC++FPGA/Tutorials/Features/scheduler_target_fmax/src/scheduler_target_fmax.cpp

```diff
@@ -0,0 +1,123 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities/include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+using namespace sycl;
+
+constexpr unsigned seed = 1313;
+
+// Forward declare the kernel names in the global scope.
+// This FPGA best practice reduces name mangling in the optimization reports.
+class NoSchedulerTargetFMAX;
+class SchedulerTargetFMAX;
+
+// Runs the Kernel
+void KernelRun(size_t size, const std::vector<char> &input_data,
+               std::vector<unsigned> &output_data) {
+
+#if defined(FPGA_EMULATOR)
+  ext::intel::fpga_emulator_selector device_selector;
+#else
+  ext::intel::fpga_selector device_selector;
+#endif
+
+  try {
+    // create the SYCL device queue
+    queue q(device_selector, dpc_common::exception_handler,
+            property::queue::enable_profiling{});
+
+    buffer input_buffer(input_data);
+    buffer output_buffer(output_data);
+
+    auto e_g = q.submit([&](handler &h) {
```

Comment:
Why are you saving the  result? We never use it.

Reply:
Do you think calculating throughput is useful in this tutorial? That's one thing I can add and it uses the returned event instance.

Reply:
Hmmm, it could be, yeah. If you can demonstrate the gains you'd see.